src/vulkan.jl

```diff
@@ -0,0 +1,43 @@
+using VulkanCore
+
+"""
+    VulkanSupported()
+Return whether the Vulkan loader has been found.
+"""
+VulkanSupported() = ccall((:glfwVulkanSupported, lib), Cint, ())
+
+"""
+    GetRequiredInstanceExtensions() -> (count, ptr, A)
+Return a 3-element tuple (count, ptr, A) where `A` is an array of names of Vulkan
+instance extensions required by GLFW for creating Vulkan surfaces for GLFW windows;
+`count` is the number of the extensions(the value of `enabledExtensionCount`);
+`ptr` is the pointer of those strings(the value of `ppEnabledExtensionNames`).
+"""
+function GetRequiredInstanceExtensions()
+    count = Ref{Cuint}(0)
+    ptr = ccall((:glfwGetRequiredInstanceExtensions, lib), Ptr{Ptr{Cchar}}, (Ref{Cuint},), count)
+    return count[], ptr, unsafe_string.(unsafe_wrap(Array, ptr, count[]))
```

Comment:
Returning `count[]` seems unnecessary since the caller can use `length(A)`. It's not clear to me why `ptr` needs to be returned either. Does it get passed into some future Vulkan function? If not, it seems sufficient to just return the array.

Reply:
I did this because the original GLFW API [`glfwGetRequiredInstanceExtensions`](http://www.glfw.org/docs/latest/group__vulkan.html#ga1abcbe61033958f22f63ef82008874b1) has an input argument `uint32_t * count`. I agree with you this is redundant and not very Julian. As for `ptr`, it will be directly passed as a field(`ppEnabledExtensionNames`) to Vulkan's [VkInstanceCreateInfo](https://github.com/JuliaGPU/VulkanCore.jl/blob/25b085baa7ccc53d3ad0b3c43854f5fca87f7042/gen/api/vk_common_1.0.0.jl#L1160) struct. Here, `ppEnabledExtensionNames` is of type `Ptr{Cstring}`, I don't know how to recover `ptr` from the `unsafe_string`ed array. I've tried:
```julia
#_, _, extensions = GetRequiredInstanceExtensions()
Ptr{Cchar}[pointer(extensions)]  # cannot convert to Ptr{Cstring}
Ptr{Cstring}(pointer(extensions))  # `VkInstanceCreateInfo` works, but `vkCreateInstance` fails
Ptr{Ptr{Cchar}}(pointer(extensions)) # `VkInstanceCreateInfo` works, but `vkCreateInstance` fails
pointer(Vector{UInt8}(string(extensions...)))  # `VkInstanceCreateInfo` works, but `vkCreateInstance` crashes
```
It easier to get the string array from `ptr` than the other way around, so this function should at least return `ptr`. With current API, the code looks like this: 
```julia
sType = vk.VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO
flags = UInt32(0)
pApplicationInfo = Base.unsafe_convert(Ptr{vk.VkApplicationInfo}, appInfo)
enabledExtensionCount, ppEnabledExtensionNames, extensions = GetRequiredInstanceExtensions()
enabledLayerCount = UInt32(0)
ppEnabledLayerNames = C_NULL
createInfo = Ref(vk.VkInstanceCreateInfo(sType, C_NULL, flags, pApplicationInfo, enabledLayerCount, ppEnabledLayerNames, enabledExtensionCount, ppEnabledExtensionNames))
instance = Ref{vk.VkInstance}(C_NULL)
result = vk.vkCreateInstance(createInfo, C_NULL, instance)
```

Reply:
@SimonDanisch  what's your opinion about this change? I think if it's not easy to recover `ptr` from the String array, it's better just return it.

Reply:
Conceptually, `glfwGetRequiredInstanceExtensions` is returning a list of strings, so I think that's what GLFW.jl should return, and it should be the responsibility of the Vulkan package to accept a list of strings.

Reply:
that makes sense. I'll do the change tonight.